src/login/pam_systemd.c

```diff
@@ -211,26 +218,45 @@ static int export_legacy_dbus_address(
         return r;
 }
 
+static int append_session_cg_limit(sd_bus_message *m, const char *limit, const char *name) {
```

Comment:
i think "append_session_cg_weight()" would be a better name, after all you don't parse arbitrary cg limits here, but weights specifically.